backend/src/test/java/service/LoginUserTest.java

```diff
@@ -0,0 +1,221 @@
+import static org.mockito.Mockito.*;
+
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class LoginUserTest {
+    
+    @InjectMocks
+    private LoginUserService loginUserService;
+
+    @Mock
+    private LoginUserMapper loginUserMapper;
+
+    @Mock
+    private LoginUserDao loginUserDao;
+
+    @Mock
+    private IncorrectLoginsService incorrectLoginsService;
+
+    @Mock
+    private Util util;
+
+    @Before
+    public void setup(){
+        MockitoAnnotations.initMocks(this);
+    }
+    
+    // test list, save, update, get, delete
+
+    // testing list
+
+    @Test
+    /**
+     * Preconditions: Login list is empty
+     * Post-conditions: Returns an empty list
+     */
+    public void testListEmpty() {
+        List<LoginUser> loginUsers = new ArrayList<>();
+        when(loginUserDao.list()).thenReturn(loginUsers);
+
+        List<LoginUserDto> result = loginUserService.list();
+
+        assertTrue(result.isEmpty());
+    }
+
+    @Test
+    /**
+     * Preconditions: Login list is not empty
+     * Post-conditions: Returns a list of LoginDto objects
+     */
+    public void testList() {
+        List<LoginUser> loginUsers = new ArrayList<>();
+
+        loginUsers.add(
+            mockLoginUser(
+                mockUser(
+                    "Michael",
+                    "Scott",
+                    "PrisonMike"
+                ),
+                "ThisIsAPassword"
+            )
+        );
+        loginUsers.add(
+            mockLoginUser(
+                mockUser(
+                    "Dwight",
+                    "Schrute",
+                    "BeetLover73"
+                ),
+                "thisTooIsAPassword"
+            )
+        );
+
+        when(loginUserDao.list()).thenReturn(loginUsers);
+
+        List<LoginUserDto> expected = loginUserMapper.model2Dto(loginUsers, new ArrayList<LoginUserDto>());
+        List<LoginUserDto> result = loginUserService.list();
+
+        assertEquals(expected, result);
+    }
+
+    @Test
+    /**
+     * Preconditions: LoginUser does not exist in the database
+     * Post-conditions: new LoginUser is saved successfully
+     */
+    public void testSave(){
+        LoginUser loginUser = mockLoginUser(
+            mockUser(
+                "Dan",
+                 "Manly",
+                  "The Man"
+            ), 
+             "password"
+        );
+        loginUser.setId(new Long("123456789"));
+
+        LoginUserDto loginUserDto = mockLoginUserDto();
+
+        Serializable expected = new Long("123456789");
+
+        // expected.put(BeapEngineConstants.SUCCESS_STR, true);
+        // expected.put("loginUserDto", loginUserDto);
+        // expected.put("status_code", HttpStatus.CREATED.value());
+
+        when(loginUserDao.get(loginUser.getId())).thenReturn(null);
+        when(loginUserMapper.dto2Model(loginUserDto, new LoginUser())).thenReturn(loginUser);
+        when(loginUserDao.save(loginUser)).thenReturn(new Long("123456789"));
```

Comment:
Easier way of returning a long is appending an L at the end

```suggestion
        when(loginUserDao.save(loginUser)).thenReturn(123456789L);
```